recipes/self-managed-otlp-ingest/instrumentation.yaml

```diff
@@ -0,0 +1,46 @@
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: opentelemetry.io/v1alpha1
+kind: Instrumentation
+metadata:
+  namespace: default
+  name: sample-java-auto-instrumentation
+spec:
+  exporter:
+    endpoint: http://otel-collector:4317
+  propagators:
+    - tracecontext
+    - baggage
+    - b3
+  sampler:
+    type: always_on
+
+  java:
+    resources:
+      limits:
+        memory: 512Mi
+      requests:
+        memory: 512Mi
+    env:
+      - name: OTEL_EXPORTER_OTLP_PROTOCOL
+        value: grpc
+      - name: OTEL_METRICS_EXPORTER
+        value: none
+      - name: OTEL_LOGS_EXPORTER
+        value: none
+      - name: OTEL_EXPORTER_OTLP_TRACES_ENDPOINT
+        value: http://opentelemetry-collector.opentelemetry.svc.cluster.local:4317
+      - name: OTEL_EXPORTER_OTLP_COMPRESSION
+        value: gzip
```

Comment:
Is this recommended?  It will increase CPU usage for the java agent and collector, but will decrease now much goes over the network.  Usually isn't recommended unless you are going outside of the cluster.

Reply:
No, it is not recommended. This was more to ensure that gzip compression works. I forgot to remove it.